### PR TITLE
limit the exponential PTO backoff to 60s

### LINKED ISSUE #3594: No limit to exponential growth of PTO
If I have a network outage lasting, for example, 1 minute, the PTO seems to grow to a similar size in that time due to exponential backoff so that when the outage clears it will still wait 1-2 minutes after the network recovers to resume communication. I think this is due to the code doing (1 << h.ptoCount) in sentPacketHandler's getPTOTimeAndSpace() and there being no limit to ptoCount.

Can there be a configured maximum backoff? Or am I looking in the wrong place? (Or is this precluded by the spec?)

### PR DESCRIPTION
**Pull request by cliffc-spirent**
resolves #3594

### CODE REVIEW

**Review by marten-seemann — COMMENTED** [review]
Can we have a test please? :)

**Review by marten-seemann — APPROVED** [review]
LGTM. We need the CLA before we can merge this though.

**Comment by marten-seemann** [comment]
@cliffc-spirent Friendly reminder that we still need the CLA.

**Comment by cliffc-spirent** [comment]
@marten-seemann the current status is that the Chief Architect has met with the General Counsel and there are some more internal forms to fill out before the CEO gets to make final approval. I wish I was kidding.

**Comment by cliffc-spirent** [comment]
CLA was submitted Friday, still waiting to hear back from Google.

**Comment by marten-seemann** [comment]
@lucas-clemente can you please check the CLA?

**Comment by lucas-clemente** [comment]
Forced a rescan, but nothing changed.

**Comment by cliffc-spirent** [comment]
It looks like the CLA finally made its way through.

File: internal/ackhandler/sent_packet_handler.go [hunk 1]
```diff
@@ -457,6 +459,14 @@ func (h *sentPacketHandler) getLossTimeAndSpace() (time.Time, protocol.Encryptio
 	return lossTime, encLevel
 }
 
+func (h *sentPacketHandler) getScaledPTO(includeMaxAckDelay bool) time.Duration {
+	pto := h.rttStats.PTO(includeMaxAckDelay) << h.ptoCount
+	if (pto > maxPTODuration) || (pto <= 0) {
```
Comment (marten-seemann) [inline internal/ackhandler/sent_packet_handler.go:464]:
```suggestion
	if pto > maxPTODuration || pto <= 0 {
```

File: internal/ackhandler/sent_packet_handler.go [hunk 2]
```diff
@@ -23,6 +23,8 @@ const (
 	amplificationFactor = 3
 	// We use Retry packets to derive an RTT estimate. Make sure we don't set the RTT to a super low value yet.
 	minRTTAfterRetry = 5 * time.Millisecond
+	// PTO duration is exponential but truncated to the following maximum as allowed by RFC 8961 Section 4.4.
```
Comment (marten-seemann) [inline internal/ackhandler/sent_packet_handler.go:26]:
```suggestion
	// The PTO duration uses exponential backoff, but is truncated to a maximum value, as allowed by RFC 8961, section 4.4.
```

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**